### DISCUSSION TITLE
Koreader with a round screen corner!

### DISCUSSION THREAD

**Discussion by Zeyadas** [koreader/koreader#8670] [Q&A]
The title is not clear but I'll try to describe it.
I bought a new phone (Samsung galaxy A02), the screen corners is not (Right angle) it's round,
![Screenshot_20220115-213321_Samsung Internet](https://user-images.githubusercontent.com/70109540/149633727-4e982807-ec7c-4996-9e1c-26246250fb84.jpg)

Now the edges of progress bar in koreader is (hidden) under the round corner but the screenshot is not showing the rounds it's a regular rectangle.

Now this can be solved by making the margins bigger, but I couldn't find where to change it, the default value is 16.

![Screenshot_20220115-214618](https://user-images.githubusercontent.com/70109540/149634169-b74767aa-0deb-475d-b4ad-ff15a34d15ce.jpg)

**Comment by NiLuJe** [comment]
There's always the patch.lua approach to tweak the viewport. I'm sure we've mentioned this somewhere in the past, because I remember doing it for my phone :).

**Comment by pazos** [comment]
See https://github.com/koreader/koreader/pull/6260.

It used to work just fine. Not sure if material guidelines no longer apply or what happened in between.

**Reply by Zeyadas** [reply]
I'm not an expert, didn't know how to fix it on my phone.
Is there any easier way to do that? An option in the Koreader app itself.

**Reply by pazos** [reply]
could take a screenshot with your phone (i.e: not in koreader)?

I doubt samsung went against guidelines too ~~match~~ much. It should be a matter of a single pixel

**Reply by poire-z** [reply]
Could the computed margin be wrong if the user is _not_ using `Screen > Screen DPI > [ ] Auto DPI` ?
Is this your case @Zeyadas ?

**Reply by Zeyadas** [reply]
OK.
This is a screenshot taken with the new phone (the Samsung A02)
![Screenshot_20220117-220826](https://user-images.githubusercontent.com/70109540/149825955-ba15e9c8-d200-4676-9069-236300b6d1a0.jpg)

As you can see it's a regular rectangle screen.
And in the same screenshot you can see that the DPI is set to auto.
And this is a screenshot (not in koreader), and it's a regular rectangle also. 

![Screenshot_20220117-220846_One UI Home](https://user-images.githubusercontent.com/70109540/149826331-94309cf3-4706-4e47-971d-4073a675d714.jpg)


Now what I meant is this picture,

![IMG_20220117_221213_352](https://user-images.githubusercontent.com/70109540/149826125-fb3e5746-4c94-42cf-9ce0-cc1e3f474a4f.jpg)

**Reply by Frenzie** [reply]
Strange. On mine it works fine. I think the roundness is comparable.
![Screenshot_2022-01-17-20-35-14-14_1aff61e6bd981aaa4ba9d0245eb6c04f](https://user-images.githubusercontent.com/202757/149828302-c301e471-37ea-40cf-bfe7-97c6dd505f79.jpg)

**Reply by Zeyadas** [reply]
> Strange. On mine it works fine. I think the roundness is comparable.

That's exactly what I want.. More margin for the progress bar, so it will be out of the (roundness).
The margins I have is 16 and I couldn't change it.
![Screenshot_20220118-190441](https://user-images.githubusercontent.com/70109540/149973834-d9cb9588-6da8-4e51-a908-fcfa0bc2d016.jpg)

**Reply by pazos** [reply]
> That's exactly what I want.. More margin for the progress bar, so it will be out of the (roundness).
The margins I have is 16 and I couldn't change it.

The margins are 16dp indeed. The value you get depends on the screen DPI. On mine its 32 because it is high density.

These values (16dp) are intended for all phones with rounded corner per android guidelines. It is strange to see a device where they don't fit.

I'm not against a new option to set a manual margin but IMO defaults are cool as they are.

In the mean time you can play with "bottom margin" value until you raise your status bar too far away from rounded corners :)

**Reply by Zeyadas** [reply]
I'll do that, thanks.

**Comment by NiLuJe** [comment]
Still does over here (Android 11).
